### ISSUE TITLE
When training to epoch [30/100], "loss:inf" appears.

### ISSUE THREAD

**Issue by xxuantgo** [rami0205/NGramSwin#9]
Hello, I'm trying to reproduce your code on the cloud server. This is the command I input in the command line. In addition, I set the default value of `args.total_epochs` to **100**.
`python3 dp_main.py --device cuda:0 --num_device 3 --model_name NGswin --batch_size 64 --target_mode light_x2`

I don't know why the “loss:inf”--When training to epoch [30/100], "loss:inf" appears. This is a screenshot of the files in the ../log folder.
<img width="493" alt="446a8955b5f76fceb895fb87ae188f1" src="https://github.com/rami0205/NGramSwin/assets/157219491/800956e3-bb12-414e-8270-9d25605dca57">

Could you please help me? Thank you!

**Comment by rami0205** [comment]
When I trained our NGswin method I didn't met this problem, while I experienced the same problem (inf loss) as training other models (such as SwinIR large model, ELAN-light, or Uformer). Unfortunately, although I have struggled to analyze this issue, I could not find fundamental reason.

So I recommend some methods to avoid it:

1. if loss function outputs inf, do not run backward process and optimize parameters (i.e., loss.backward() and optimizer.step()).
2. change learning rate (decrease or increase)

Thanks.
